internal/vault/config_test.go

```diff
@@ -72,6 +72,17 @@ func TestMakeVaultClient(t *testing.T) {
 			CACert:        nil,
 			expectedError: nil,
 		},
+		"headers": {
```

Comment:
May want to add a test to ensure proper handling of the Vault namespace config e.g. `cfg.VaultNamespace` takes precedence over `X-Vault-Namespace`